### PR TITLE
[asr->fortran] implement visit_Array{Transpose, Pack, All}

### PR DESCRIPTION
**Pull request by khushi-411**

### CODE REVIEW

File: src/libasr/codegen/asr_to_fortran.cpp [hunk 1]
```diff
@@ -1232,7 +1256,15 @@ class ASRToFortranVisitor : public ASR::BaseVisitor<ASRToFortranVisitor>
...
 
-    // void visit_ArrayAll(const ASR::ArrayAll_t &x) {}
+    void visit_ArrayAll(const ASR::ArrayAll_t &x) {
+        std::string r;
+        r += "all";
+        r += "(";
+        visit_expr(*x.m_mask);
+        r += s;
+        r += ")";
+        s = r;
```
Comment (Shaikh-Ubaid) [inline src/libasr/codegen/asr_to_fortran.cpp:1266]: How about?

```suggestion
        visit_expr(*x.m_mask);
        s = "all(" + s + ")";
```
This seems shorter and more readable.

File: src/libasr/codegen/asr_to_fortran.cpp [hunk 2]
```diff
@@ -1216,9 +1216,33 @@ class ASRToFortranVisitor : public ASR::BaseVisitor<ASRToFortranVisitor>
...
 
-    // void visit_ArrayTranspose(const ASR::ArrayTranspose_t &x) {}
+    void visit_ArrayTranspose(const ASR::ArrayTranspose_t &x) {
+        std::string r;
+        r += "transpose";
+        r += "(";
+        visit_expr(*x.m_matrix);
+        r += s;
+        r += ")";
+        s = r;
```
Comment (Shaikh-Ubaid) [inline src/libasr/codegen/asr_to_fortran.cpp:1226]: Similarly here,
```suggestion
        visit_expr(*x.m_matrix);
        s = "transpose(" + s + ")";
```

File: src/libasr/codegen/asr_to_fortran.cpp (hunk 1 above)
Comment (khushi-411) [reply src/libasr/codegen/asr_to_fortran.cpp]: Great idea; I updated your suggestion for `transpose`. But I think `all` also supports the `dim` parameter, so I updated the code accordingly.

File: src/libasr/codegen/asr_to_fortran.cpp [hunk 3]
```diff
@@ -1232,7 +1251,20 @@ class ASRToFortranVisitor : public ASR::BaseVisitor<ASRToFortranVisitor>
...
+    void visit_ArrayAll(const ASR::ArrayAll_t &x) {
+        std::string r;
+        r += "all";
+        r += "(";
+        visit_expr(*x.m_mask);
+        r += s;
+        if (x.m_dim) {
+            r += "[, ";
+            visit_expr(*x.m_dim);
+            r += "]";
```
Comment (Thirumalai-Shaktivel) [inline src/libasr/codegen/asr_to_fortran.cpp:1263]:
```suggestion
            visit_expr(*x.m_dim);
            r += s;
```

**Comment by certik** [comment]
The test failure was due to https://github.com/lfortran/lfortran/issues/2776. I rebased it, let's see if things pass.

**Review by Thirumalai-Shaktivel — APPROVED** [review]
Looks great, thank you!

**Comment by khushi-411** [comment]
Thanks for all your reviews, @certik, @Shaikh-Ubaid, @Thirumalai-Shaktivel!!

**Pull request merged**